### PR TITLE
Document overridable YAML configuration

### PR DESCRIPTION
**Pull request by Cubxity**
This documents changes in the following PR:
https://github.com/PaperMC/Paper/pull/6710.

### CODE REVIEW

File: source/server/configuration.rst [hunk 1]
```diff
@@ -22,6 +22,28 @@ their respective documentation pages.
 
     Last updated Sep 3rd, 2021 for MC 1.17.1, Paper build #249
 
+Overriding settings
+===================
+
+Paper introduced environment variables to override configuration values in `paper.yml`, `spigot.yml`, and `server.properties`.
+The environment variable name starts with the prefix (see the table below), following by the configuration key (``.`` and ``-`` is replaced with ``_``).
```
Comment (NoahvdAa) [inline source/server/configuration.rst:29]:
```suggestion
The environment variable name starts with the prefix (see the table below), followed by the configuration key (``.`` and ``-`` are replaced with ``_``).
```

File: source/server/configuration.rst [hunk 2]
```diff
@@ -22,6 +22,28 @@ their respective documentation pages.
 
     Last updated Sep 3rd, 2021 for MC 1.17.1, Paper build #249
 
+Overriding settings
+===================
+
+Paper introduced environment variables to override configuration values in `paper.yml`, `spigot.yml`, and `server.properties`.
+The environment variable name starts with the prefix (see the table below), followed by the configuration key (``.`` and ``-`` are replaced with ``_``).
+This works with string, int, double, and boolean values.
```
Comment (sulu5890) [inline source/server/configuration.rst:30]: I'm not really even sure if this is better, but something about how this is worded feels odd. Let me know what you think.

```suggestion
Paper has introduced environment variables to override configuration values in `paper.yml`, `spigot.yml`, and `server.properties`.
Each environment variable starts with a prefix (see the table below), followed by the configuration key, replacing ``.`` and ``-`` with ``_``.
This works with string, int, double, and boolean values.
```

**Review by e-im — APPROVED** [review]

File: source/server/configuration.rst [hunk 3]
```diff
@@ -22,6 +22,55 @@ their respective documentation pages.
...
+Overriding settings
+===================
+
+Paper has introduced environment variables and JVM properties to override configuration values in `paper.yml`, `spigot.yml`, `bukkit.yml`, and `server.properties`.
+
+Environment variables
+~~~~~~~~~~~~~~~~~~~~~
+
+Each environment variable starts with a prefix (see the table below), followed by the configuration key, replacing ``.`` and ``-`` with ``_``.
+This works with string, int, boolean, double, and long values.
```
Comment (lynxplay) [inline source/server/configuration.rst:34]: Do you think this is enough to communicate that overriding may not be used on anything complex like maps or lists ?

Comment (Cubxity) [reply source/server/configuration.rst:34]: I think I may have to clarify it a bit more

**Review by e-im — COMMENTED** [review]
~~Even though the name is quite meh as they aren't system-wide at all, for consistency, call them `System Properties` rather than `JVM Properties`.~~ fine as is

**Comment by e-im** [comment]
Thank you for writing docs for your Paper PR. We are in the process of [moving to new documentation](https://github.com/PaperMC/docs/issues/12). There is no need to do anything now; however, if your Paper PR is accepted, please open a new pr to https://github.com/PaperMC/docs if you are able. I am also happy to take care of this for you when the time comes.

**Pull request closed without merging**